pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -732,20 +727,25 @@ func (m *kubeGenericRuntimeManager) doPodResizeAction(pod *v1.Pod, podStatus *ku
 			result.Fail(err)
 			return
 		}
-		if currentPodMemoryUsage >= uint64(*podResources.Memory) {
+		if podResources.Memory != nil && currentPodMemoryUsage >= uint64(*podResources.Memory) {
 			klog.ErrorS(nil, "Aborting attempt to set pod memory limit less than current memory usage", "pod", pod.Name)
 			result.Fail(fmt.Errorf("Aborting attempt to set pod memory limit less than current memory usage for pod %s", pod.Name))
 			return
 		}
-		if errResize := resizeContainers(v1.ResourceMemory, int64(*currentPodMemoryConfig.Memory), *podResources.Memory, 0, 0); errResize != nil {
+		var currentMemory, desiredMemory int64
+		if podResources.Memory != nil {
+			currentMemory = *currentPodMemoryConfig.Memory
+			desiredMemory = *podResources.Memory
+		}
+		if errResize := resizeContainers(v1.ResourceMemory, currentMemory, desiredMemory, 0, 0); errResize != nil {
 			result.Fail(errResize)
 			return
 		}
 	}
 	if len(podContainerChanges.ContainersToUpdate[v1.ResourceCPU]) > 0 || podContainerChanges.UpdatePodResources {
-		if podResources.CPUQuota == nil || podResources.CPUShares == nil {
-			klog.ErrorS(nil, "podResources.CPUQuota or podResources.CPUShares is nil", "pod", pod.Name)
-			result.Fail(fmt.Errorf("podResources.CPUQuota or podResources.CPUShares is nil for pod %s", pod.Name))
+		if podResources.CPUShares == nil {
```

Comment:
Looking at https://github.com/kubernetes/kubernetes/blob/6bc0768c62b9dade91b22d1bfcc115edf1f0ccad/pkg/kubelet/cm/helpers_linux.go#L120, this should not be possible, right? If so, leave a comment to that effect.